src/commands/kv/key/put.rs

```diff
@@ -82,3 +72,78 @@ pub fn put(target: &Target, user: &GlobalUser, data: KVMetaData) -> Result<(), f
 
     Ok(())
 }
+
+fn get_response(
+    data: KVMetaData,
+    user: &GlobalUser,
+    url: &Url,
+) -> Result<reqwest::blocking::Response, failure::Error> {
+    let url_into_str = url.to_string();
+    let client = http::legacy_auth_client(user);
+    let res = match data.metadata {
+        Some(metadata) => {
+            let part = if data.is_file {
```

Comment:
data.is_file is being handled differently from get_request_body, which means people are going to get confusingly different errors for invalid files depending on whether they added metadata or not.

Reply:
Okay. We cannot use a multipart upload without metadata. So we can:
1. Edit `get_request_body` to always return a `String`, so errors like "file not found" are always in the same format.
2. Massage all errors from `multipart::Part::file` and `std::fs::File::open` to look the same.
3. Edit the API to allow for multipart uploads without metadata, so this code will always use `multipart::Part::file`.

I think 2 is too much work. What do you think between 1 & 3?

Reply:
I see what you're saying.
I don't think we want to change the api.
Part is just a wrapper around Body, so it seems like you should be able to use get_request_body as is, but the constructor for Part that takes a Body is private >_<
Regarding using String as lowest common denominator instead of Vec<u8>, what does the current implementation do with a file of arbitrary bytes?

Reply:
Good point, we need to respect arbitrary bytes. I used `Vec<u8>`.